src/App.tsx

```diff
@@ -1,27 +1,64 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { get5First, getAll, getRedGoods } from './api/goods';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+type SortType = 'All' | 'FirstFive' | 'OnlyRed' | '';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [sortType, setSortType] = useState<SortType>('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const sortGoods = () => {
+    switch (sortType) {
+      case 'All':
+        getAll().then(goodsFromServer => setGoods(goodsFromServer));
+        break;
+      case 'FirstFive':
+        get5First().then(goodsFromServer => setGoods(goodsFromServer));
+        break;
+      case 'OnlyRed':
+        getRedGoods().then(goodsFromServer => setGoods(goodsFromServer));
+        break;
+      default:
+        setGoods([]);
```

Comment:
let's use a callback shorthand
```suggestion
    switch (sortType) {
      case 'All':
        getAll().then(setGoods);
        break;
      case 'FirstFive':
        get5First().then(setGoods);
        break;
      case 'OnlyRed':
        getRedGoods().then(setGoods);
        break;
      default:
        setGoods([]);
```